lib/app_profiler/middleware.rb

```diff
@@ -31,7 +31,7 @@ def profile(env, params)
 
       return yield unless before_profile(env, params_hash)
 
-      profile = AppProfiler.run(params_hash) do
+      profile = AppProfiler.run(params_hash, backend: params.backend) do
```

Comment:
We should probably be changing the behaviour of `params.to_h` instead of explicitly setting the backend, right?

Reply:
I'm not sure I understand. Can't you just use `**params_hash` to have the hash satisfy the kwarg? You shouldn't need to change the signature:
```
irb(main):001* def run(*args, backend: nil, **kwargs)
irb(main):002*   puts args.inspect, backend, kwargs.inspect
irb(main):003> end
=> :run
irb(main):004> run(**{ backend: "hello", other: :opt })
[]
hello
{:other=>:opt}
=> nil
```

Reply:
yes so as you've defined it there it works fine as when you do the splat, the backend parameter is extracted into a variable that you can use for the intended purpose of toggling the backend, and the other parameters are forwarded to the actual profiler.

I had assumed by your earlier comment:

> We should probably be changing the behaviour of params.to_h instead of explicitly setting the backend, right?

That you thought we should drop that change to the **signature** too. But i guess you just meant how we generate the hash that it is called with.

That is a much simpler change to make, my apologies for the misunderstanding - i'll make the change.